### PR TITLE
Speed up set closure by using a fixed amount of memory.

### PR DESCRIPTION
**Pull request by ltratt**
In a typical description of this algorithm -- and in the previous implementation
-- one has a todo set which contains pairs (prod_i, dot). Unfortunately this is
a slow way of doing things. Searching the set for the next item and removing it
is slow; and, since we don't know how many potential dots there are production,
the set is of potentially unbounded size, so we can end up resizing memory.
Since this function is the most expensive in the table generation, using a
HashSet (which is the "obvious" solution) is pretty slow.

However, we can reduce these costs through two observations:
  1) The initial todo set is populated with (prod_i, dot) pairs that all come
     from self.items.keys(). There's no point copying these into a todo list.
  2) All subsequent todo items are of the form (prod_off, 0). Since the dot in
     these cases is always 0, we don't need to store pairs: simply knowing which
     prod_off's we need to look at is sufficient. We can represent these with a
     fixed-size bitfield.
All we need to do is first iterate through the items in 1 and, when it's
exhausted, continually iterate over the bitfield from 2 until no new items have
been added.

On my machine, this speeds up the time needed to build the PHP LR table by 10%,
from 0.33s to 0.30s.

### CODE REVIEW

File: src/lib/stategraph.rs [hunk 1]
```diff
@@ -199,22 +199,58 @@ impl Itemset {
...
-        for &(prod_i, dot) in self.items.keys() {
-            todo.push((prod_i, dot));
-        }
+        // dissertation.
+
+        let mut new_is = self.clone(); // The new itemset we're building up
+
+        // In a typical description of this algorithm, one would have a todo set which contains
+        // pairs (prod_i, dot). Unfortunately this is a slow way of doing things. Searching the set
+        // for the next item and removing it is slow; and, since we don't know how many potential
```
Comment (ptersilie) [inline src/lib/stategraph.rs]: ```dots there are in a production```

**Comment by ltratt** [comment]
If the new commit fixes things for you, let me know, I'll squash, then we'll be in a state to merge (hopefully).

**Comment by ptersilie** [comment]
Yep, looks good.

**Comment by ltratt** [comment]
OK, ready for merging (with luck).

**Comment by ptersilie** [comment]
Merged.

**Pull request merged**